### ISSUE TITLE
Microstrip dispersion pipeline

### ISSUE THREAD

**Issue by gvcallen** [paramrf/paramrf#62]
Part of #59

`area: models/lines` · `size: L` · `breaking` · `depends: #61` · `correctness`

### Context

Two things land here: the modal-dispersion axis that microstrip has been missing,
and the remainder of a correctness gap in how microstrip reaches its RLGC values.

**#61 has landed since this was drafted, and it moved the goalposts.** The
microstrip path no longer builds per-unit-length values from lossless L and C:
`QuasiStaticResult.to_immittance` now derives $Z$ and $Y$ directly from $Z_c$ and
a complex $\varepsilon_e$, and the `epr - 1` division that caused the G
singularity is gone. Sections below marked **(partly landed)** describe work that
is now smaller than written; re-measure before treating their error tables as a
failing baseline. The dispersion axis itself is untouched and is the bulk of this
ticket.

### Split modal dispersion onto its own axis (D8 in #59)

Material dispersion and modal dispersion are different physics and must not share
a field. Modal dispersion is the concentration of field into the substrate as
frequency rises — it happens even with a perfectly non-dispersive material.
Crucially it exists only for **inhomogeneous** cross-sections: microstrip and CPW
have it; coax and stripline do not, because they are homogeneously filled. Encode
that by simply not giving coax and stripline a `dispersion` field.

The three-stage pipeline, mirroring `skrf.media.mline.MLine.__init__`:

1. **Material** — `dielectric.epsilon_r(f)` gives complex εr(f)
2. **Quasi-static** — `formulation` maps (w, h, t, εr(f)) to εeff(0), Zc(0), w_eff
3. **Modal dispersion** — `dispersion` maps (εeff(0), Zc(0), εr, w, w_eff, h, t, f)
   to εeff(f), Zc(f)

Stages 1 and 2 exist on `main` today as `AbstractDielectric.epsilon_r` and
`AbstractMicrostripFormulation.quasi_static` -> `QuasiStaticResult`. Stage 3 is new.

```python
class AbstractMicrostripDispersion(Module):
    def run(self, freq, *, eps_eff_0, zc_0, eps_r,
            w, w_eff, h, t) -> tuple[Array, Array]: ...
```

- Implement `KirschningJansen` first; `Kobayashi` and `Yamashita` as follow-ups.
- `MicrostripLine` gains `dispersion: AbstractMicrostripDispersion | None`.
- Default to `KirschningJansen()`, matching scikit-rf and ADS. This changes
  existing microstrip results — unlike the dielectric default in D7, this is an
  accuracy question, not a causality one. Flag it in the changelog.
- `dispersion=None` disables stage 3 entirely.
- Add `HammerstadJensenMicrostripFormulation` alongside Wheeler while in here —
  Kirschning–Jansen is normally paired with it.

### Derive RLGC by exact inversion **(partly landed in #61)**

As drafted, this section described `WheelerMicrostripSolver` building lossless L
and C and bolting R and G on afterwards, so that
`AbstractRLGCLine.zc_and_gammaL` recomputed a $Z_c$ that no longer equalled the
one the formulation produced. That class is now
`WheelerMicrostripFormulation`, the base is now `AbstractImmittanceLine`, and
`QuasiStaticResult.to_immittance` builds

$$Z = \frac{j\omega Z_c \sqrt{\varepsilon_e}}{c} + \frac{2 Z_s}{W_{eff}}
\qquad
Y = \frac{j\omega \sqrt{\varepsilon_e}}{Z_c c}$$

so the dielectric half of the round trip is already exact. The error table below
was measured on `main` **before** #61 and is retained only as the historical
motivation; it is not the current baseline.

| Case | \|Zc\| error | β error |
|------|------------|---------|
| FR-4, tand = 0.02 | 0.46 – 0.87 % | 0.004 % |
| High-loss laminate, tand = 0.05 | 1.8 – 2.2 % | 0.025 % |
| Thin lossy trace, 0.2 mm, rho = 1e-6 | **0.5 – 17.9 %** | 1.6 % |

**What remains is the stage-3 output.** Kirschning–Jansen yields Zc(f) and γ(f)
directly, and those must reach `ImmittanceResult` by exact inversion rather than
by any lossless construction. The relation is an exact per-frequency bijection,
four real numbers to four real numbers:

```
Z = R + 1j*w*L = gamma * zc
Y = G + 1j*w*C = gamma / zc
```

The branch choice is unambiguous: for passive parameters both `sqrt` arguments
lie in the upper half-plane, so the principal branch already gives Re(γ) ≥ 0 and
Re(Zc) > 0.

- Stage 3 yields Zc(f) and γ(f); `MicrostripLine` inverts them into the
  `ImmittanceResult` form that #61 introduced.
- Keep `MicrostripLine` on `AbstractImmittanceLine`. The RLGC view stays
  available for plotting and introspection via `ImmittanceResult.R`/`L`/`G`/`C`.
- `CoaxialLine` is unaffected — there R, L, G, C are primary and the round trip is
  already exact.
- Add a passivity check after inversion: Re(Z) >= 0 and Re(Y) >= 0. A negative
  value means the empirical formulas produced a non-physical (Zc, γ) in that
  regime, worth surfacing rather than silently propagating.

### The G singularity as εr -> 1 **(landed in #61 — hold the regression test)**

As drafted, `WheelerMicrostripSolver.run` computed
`G = (1/(Za*c)) * (epr*(epe-1)/(epr-1)) * tand * freq.w`, dividing by `epr - 1`
at exactly the point where `MicrostripLine.epr` carries `GreaterThan(1.0)` — a
singularity sitting on the constraint boundary.

That form is gone: complex εr now flows through and the dielectric loss falls out
of Im(εeff) with no filling-factor division. **No fix is needed.** Keep the
acceptance criterion below as a regression test so a future dispersion or
convention change cannot reintroduce the division.

### Expose the real-versus-complex permittivity convention (D6 in #59)

Complex εr already propagates end to end after #61, which is the `'complex'`
half of this decision. What remains is the switch.

scikit-rf maintains both behind `compatibility_mode` because QUCS and ADS
genuinely disagree.

- **complex** (ADS-like, default) — full εr(f) through both stages; loss falls out
  of Im(εeff). Internally consistent, branch-free, and the only path that stays
  correct when Im(εr) is not small. This is today's behaviour.
- **real** (QUCS-like) — `real(eps_r(f))` into the formulas, loss via a separate
  attenuation path. Defensible because these formulas were fitted to real-εr EM
  data, making complex input an unvalidated extrapolation.

Add a static field `epsilon_convention: 'complex' | 'real'`, default `'complex'`.
Guard Hammerstad's `b = 0.564 * ((eps_r - 0.9)/(eps_r + 3))**0.053` — a complex
power with a branch cut.

### Acceptance criteria

- `dispersion=None` reproduces #61 behaviour exactly.
- `KirschningJansen` matches `skrf.media.MLine(compatibility_mode=None)` to within
  0.1 % for εr in 2.2–10, W/H in 0.1–10, up to 50 GHz.
- `CoaxialLine` has no `dispersion` field.
- Regression test on a high-loss microstrip asserting `line.zc(freq)` equals the
  dispersion stage's Zc(f) to 1e-12, and `imag(gamma)` equals (ω/c)·sqrt(εeff) to
  1e-12. **Write this first** and record what it actually does on `main` after
  #61 — if it already passes, say so in the test rather than assuming the
  pre-#61 margins still hold.
- `epr = 1.0 + 1e-12` gives finite, continuous G; `jax.grad` of `s` wrt
  permittivity is finite across εr in [1, 12]. (Expected to pass on arrival — it
  is a regression guard, not a fix.)
- `'real'` matches `compatibility_mode='qucs'`; `'complex'` matches
  `compatibility_mode=None`.
- Any change to microstrip results is a correction, not a regression — note it in
  the changelog rather than loosening a test.

**Comment by gvcallen** [comment]
Compatibility notes from #61 (PR #66), which has now landed the stage 1/stage 2 seam this issue builds on.

## Still accurate

The stage-2 → stage-3 handoff lines up: `QuasiStaticResult` carries `eps_eff`, `zc` and `w_eff`, which are exactly the three stage-2 outputs `AbstractMicrostripDispersion` wants; `eps_r`, `w`, `h` and `t` the line already holds. `CoaxialLine` has no `dispersion` field and now has no reason to grow one. `dispersion=None` reproducing Issue 2 behaviour is trivially satisfiable.

For 3d, #61 hard-wired complex εr, which is this issue's `'complex'` default, so `epsilon_convention` stays purely additive. Wheeler is safe with complex input — εeff is linear in εr, no fractional powers, no branch cut. The guard this issue asks for is only needed for Hammerstad's `b = 0.564 * ((eps_r - 0.9)/(eps_r + 3))**0.053`, which does not exist yet.

## 3c is already done

The singularity described here no longer exists. #61 replaced the filling-factor form

```python
G = (1 / (Za * c)) * (epr * (epe - 1) / (epr - 1)) * tand * freq.w
```

with a shunt admittance derived from complex εeff, `Y = 1j*w*sqrt(eps_eff)/(zc*c)`, which has no `epr - 1` denominator anywhere. That is the fix 3c proposes, arrived at as a side effect of making εr complex. Worth striking the section so it is not implemented twice.

Note the filling factor is preserved exactly, not approximated: Wheeler's εeff is linear in εr, so `(εeff − 1)/(εr − 1) = (1 + t3)/2` identically, and G agrees with the pre-refactor expression to 5e-16 relative.

## 3b will rewrite `QuasiStaticResult.to_immittance`

The one place the landed code and this issue disagree. `to_immittance` currently does

```python
Z = 1j * w * zc * sqrt_eps_eff / c + 2 * zs / w_eff
Y = 1j * w * sqrt_eps_eff / (zc * c)
```

`Y = gamma/Zc` is already the exact bijection, but `Z = gamma*Zc + 2*Zs/w_eff` bolts the conductor loss on outside it. That is structurally the pattern 3b calls a bug, and it will fail the acceptance criterion that `line.zc(freq)` equal the formulation's Zc to 1e-12 on a lossy microstrip.

It resolves cleanly under this issue's design: once stage 3 returns a Zc(f) and γ(f) that already carry conductor attenuation, the line does the pure inversion `Z = gamma*zc`, `Y = gamma/zc` and the extra series term disappears. So `to_immittance` is the right shape for a pipeline with no stage 3, and this issue should expect to replace its body rather than extend it.

Two consequences:

- The microstrip inductance deviation flagged on PR #66 (internal inductance now included, ~0.9% on L at the low end) becomes moot, since the whole series term is restructured here.
- The passivity check this issue wants, Re(Z) >= 0 and Re(Y) >= 0, has an obvious home on `ImmittanceResult`.

## Two stale references in the issue text

1. "Keep `MicrostripLine` on `AbstractRLGCLine`", and the description of `zc_and_gammaL` recomputing `zc = sqrt((R + 1j*w*L) / (G + 1j*w*C))`. That class is now `AbstractImmittanceLine` and `zc_and_gammaL` is already `sqrt(Z/Y)`. The intent — RLGC stays available for plotting and introspection — is satisfied by the derived properties on `ImmittanceResult`.
2. `AbstractMicrostripDispersion.run` inherits a verb #61 removed. `run` is the solver verb in ParamRF: every other `def run` in the package belongs to a circuit solver, an optimizer or an inferer. Formulations are now named for what they return (`AbstractCoaxialFormulation.immittance`, `AbstractMicrostripFormulation.quasi_static`), following the materials package. `disperse(...)` fits the dispersion stage, since it returns a corrected (εeff, Zc) pair rather than one named quantity.
